sklearn/linear_model/tests/test_ridge.py

```diff
@@ -551,6 +552,85 @@ def test_ridge_gcv_sample_weights(
     assert_allclose(gcv_errors, kfold_errors, rtol=1e-3)
     assert_allclose(gcv_ridge.coef_, kfold.coef_, rtol=1e-3)
     assert_allclose(gcv_ridge.intercept_, kfold.intercept_, rtol=1e-3)
+"""
+
+
+@pytest.mark.parametrize("fit_intercept", [True, False])
+def test_ridge_gcv_stored_predictions(fit_intercept):
+    # check the predictions stored in `cv_values_` are equivalent to the
+    # prediction computed with a LOO with a Ridge at a specific alpha when the
+    # `scoring` is not `None`. We check as well that the predictions are
+    # rescaled if fitted with an intercept.
+    # Partly a regression test for #13998
+    X, y = make_regression(n_samples=6, n_features=2, random_state=42)
+    ridge_cv = RidgeCV(
+        fit_intercept=fit_intercept, scoring='neg_mean_squared_error',
+        store_cv_values=True, alphas=[1.]
+    )
+    ridge_cv.fit(X, y)
+    ridge = Ridge(fit_intercept=fit_intercept)
+    loo_pred = cross_val_predict(ridge, X, y, cv=len(X))
+    assert_allclose(loo_pred.ravel(), ridge_cv.cv_values_.ravel())
+
+
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize("use_sample_weight", [True, False])
+def test_ridge_gcv_equivalence_prediction_metric(fit_intercept,
+                                                 use_sample_weight):
+    # Check the consistency between the cv_values_ obtained with and without
+    # a score. The default score being the mean squared error, we can compute
+    # the error from the prediction with `scoring='neg_mean_squared_error`
+    # the score reported when `scoring=None`
+    X, y = make_regression(n_samples=6, n_features=2, random_state=42)
+    ridge_cv_no_score = RidgeCV(
+        fit_intercept=fit_intercept, scoring=None, store_cv_values=True,
+        alphas=[1.]
+    )
+    ridge_cv_score = RidgeCV(
+        fit_intercept=fit_intercept, scoring='neg_mean_squared_error',
+        store_cv_values=True, alphas=[1.]
+    )
+
+    if use_sample_weight:
+        rng = np.random.RandomState(0)
+        sample_weight = rng.rand(X.shape[0])
+    else:
+        sample_weight = None
+    ridge_cv_no_score.fit(X, y, sample_weight=sample_weight)
+    ridge_cv_score.fit(X, y, sample_weight)
+
+    assert (
+        ridge_cv_no_score.cv_values_.mean() == pytest.approx(
+            mean_squared_error(y, ridge_cv_score.cv_values_.ravel()))
```

Comment:
From my above message, we could expect to pass `sample_weight` to `mean_squared_error`. However, the np.average will normalize by the sum of the weight. It is equivalent to normalize the weight such that they sum to 1.

However, the loss internally will not do such thing because we want to impose that 3 times a sample correspond to seeing 3 times a sample increasing the loss x3.

So this is not straightforward what to implement.

Basically, with the initial semantic the assert would be

```python
ridge_cv_no_score.cv_values_.mean() == pytest.approx(
    (((y - ridge_cv_score.cv_values_.ravel()) ** 2) * sample_weight).sum()
)
```

while the `mean_squared_error` would be equivalent to 

```python
ridge_cv_no_score.cv_values_.mean() == pytest.approx(
    (((y - ridge_cv_score.cv_values_.ravel()) ** 2) * sample_weight / sample_weight.sum()).sum()
)
```

Reply:
> we want to impose that 3 times a sample correspond to seeing 3 times a sample increasing the loss x3

indeed, otherwise it amounts to changing the regularization parameter

Reply:
> However, the loss internally will not do such thing because we want to impose that 3 times a sample correspond to seeing 3 times a sample increasing the loss x3.

Sorry but I can't understand your discussions, this PR is not related to the loss function?

Reply:
I got confused as well so please ignore my previous comment. As far as I can tell the changes made in this PR yield the correct behaviour for `_RidgeGCV`